src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -96,98 +103,245 @@ export default class Preview extends Setup {
         );
 
-        const targetAppArguments = CommandLineUtils.resolveFlag(
-            this.flags.targetapparguments,
-            ''
-        );
-
         const projectDir = CommandLineUtils.resolveFlag(
             this.flags.projectdir,
             process.cwd()
         );
 
-        const componentName = this.flags.componentname;
+        const configFile = CommandLineUtils.resolveFlag(
+            this.flags.configfile,
+            ''
+        );
+
+        const component = this.flags.componentname;
 
-        const launcher = new IOSLauncher(simName);
+        let targetAppArguments: Map<string, string> = new Map();
+        let launchActivity: string = '';
+        if (PreviewUtils.isTargetingBrowser(targetApp) === false) {
+            const json = PreviewUtils.getConfigFileAsJson(configFile);
 
-        return launcher.launchPreview(
-            componentName,
-            projectDir,
-            targetApp,
-            targetAppArguments
-        );
+            targetAppArguments = PreviewUtils.getAppLaunchArguments(
+                json,
+                platform,
+                targetApp
+            );
+
+            launchActivity = PreviewUtils.getAppLaunchActivity(json, targetApp);
+        }
+
+        const promise = CommandLineUtils.platformFlagIsIOS(this.flags.platform)
+            ? this.launchIOS(
+                  device,
+                  component,
+                  projectDir,
+                  targetApp,
+                  targetAppArguments
+              )
+            : this.launchAndroid(
+                  device,
+                  component,
+                  projectDir,
+                  targetApp,
+                  targetAppArguments,
+                  launchActivity
+              );
+
+        return promise;
     }
 
-    public launchAndroid(): Promise<boolean> {
-        const emulatorName = CommandLineUtils.resolveFlag(
-            this.flags.target,
-            androidConfig.defaultEmulatorName
-        );
+    protected async init(): Promise<void> {
+        await super.init();
+        const logger = await Logger.child('mobile:preview', {});
+        this.logger = logger;
+    }
 
-        const targetApp = CommandLineUtils.resolveFlag(
-            this.flags.targetapp,
-            PreviewUtils.BROWSER_TARGET_APP
-        );
+    protected _help(): never {
+        const isCommandHelp =
+            this.argv.filter(
+                (v) => v.toLowerCase() === '-h' || v.toLowerCase() === '--help'
+            ).length > 0;
 
-        const targetAppArguments = CommandLineUtils.resolveFlag(
-            this.flags.targetapparguments,
-            ''
-        );
+        if (isCommandHelp) {
+            super._help();
+        } else {
+            const message = `
+Below is a sample configuration:
 
-        const projectDir = CommandLineUtils.resolveFlag(
-            this.flags.projectdir,
-            process.cwd()
-        );
+{
```

Comment:
The help message should probably be moved to a "message" file so that it can be translated to other languages, but to my knowledge, "message" files are just JSON files and we can't have a 'blob' of string as a single JSON property value.

So not too sure what to do here.

Reply:
The examples I've seen embed \n in the message strings. It's not ideal, but it looks like the way to do it today.

Reply:
Hmm... that would make a very long message which is hard to maintain/edit in the future. But sure, will make the change.